frontend/src/components/core/MainMenu/MainMenu.tsx

```diff
@@ -215,8 +220,108 @@ const MenuListItem = forwardRef<HTMLLIElement, MenuListItemProps>(
   }
 )
 
-function MainMenu(props: Props): ReactElement {
+const DevMenuListItem = forwardRef<HTMLLIElement, MenuListItemProps>(
+  (
+    {
+      item,
+      "aria-selected": ariaSelected,
+      onClick,
+      onMouseEnter,
+      $disabled,
+      $isHighlighted,
+    },
+    ref
+  ) => {
+    const { label, shortcut, hasDividerAbove } = item
+    const menuItemProps = {
+      isDisabled: $disabled,
+      isHighlighted: $isHighlighted,
+      isRecording: Boolean(item.stopRecordingIndicator),
+    }
+    const interactiveProps = $disabled
+      ? {}
+      : {
+          onClick,
+          onMouseEnter,
+        }
+
+    return (
+      <>
+        {hasDividerAbove && <StyledMenuDivider />}
+        <StyledDevMenuItem
+          ref={ref}
+          role="option"
+          aria-selected={ariaSelected}
+          aria-disabled={$disabled}
+          {...menuItemProps}
+          {...interactiveProps}
+        >
+          <StyledMenuItemLabel {...menuItemProps}>{label}</StyledMenuItemLabel>
+          {shortcut && (
+            <StyledMenuItemShortcut {...menuItemProps}>
```

Comment:
I think you can get rid of a ton of code duplication here by having `MenuListItem` be a function that takes in which styled component (`StyledMenuItem` or `StyledDevMenuItem`) to use as an argument and returns a component (essentially making it a higher order component).

Then in the `SubMenu` component you can write something like
```javascript
const StyledMenuItem = isDevMenu ? StyledDevMenuItem ? StyledMenuItem
return (
  <StatefulMenu
    /* ...stuff */
    overrides={{
      Option: MenuListItem(StyledMenuItem),
      /* ...more stuff */
    }}
  />
)
```